frankenphp.c

```diff
@@ -472,7 +563,28 @@ int frankenphp_update_server_context(
   return SUCCESS;
 }
 
+PHPAPI void get_full_env(zval *track_vars_array) {
+  struct go_getfullenv_return full_env = go_getfullenv(thread_index);
```

Comment:
I think we can probably cache the environment somewhere here as a thread local (or even global) `zend_array`, so it will only get imported once. I'll check later this week again, but I think that's how FPM also behaves with  since `putenv` doesn't alter `$_ENV` or `$_SERVER`

Reply:
Should probably do it on the go side and not mix source of truths, but yeah.

Reply:
Yeah even in the manual [someone mentions](https://www.php.net/manual/de/function.putenv.php#98540) how putenv doesn't affect `$_ENV`. Since `$_ENV` is unchanging for the duration of the process I think it would be most efficient to directly store it as a `zend_array` at the start of the process.

`putenv` and `getenv` should always access the real environment though like in your implementation 👍

Reply:
Interesting behavior spot, thanks! Didn't expect it to refresh `$_ENV` repeatedly. Easy fix though.